lib/streamlit/delta_generator.py

```diff
@@ -326,6 +326,12 @@ def _main_dg(self) -> DeltaGenerator:
         """
         return self._parent._main_dg if self._parent else self
 
+    def __getstate__(self):
```

Comment:
We need these functions to allow the delta generator to be serialized by cloudpickle